Indicators/Vortex.cs

```diff
@@ -0,0 +1,132 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Data.Market;
+using QuantConnect.Indicators;
+
+/// <summary>
+/// This indicator computes the Vortex Indicator which measures the strength of a trend and its direction.
+/// The Vortex Indicator is calculated by comparing the difference between the current high and the previous low with the difference between the current low and the previous high.
+/// The values are accumulated and smoothed using a rolling window of the specified period.
+/// </summary>
+
+public class VortexIndicator : IndicatorBase<IBaseDataBar>
+{
+    private readonly int _period;
+    private readonly RollingWindow<IBaseDataBar> _window;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="VortexIndicator"/> class.
+    /// </summary>
+    /// <param name="period">The number of periods used to calculate the indicator.</param>
+    public VortexIndicator(int period) : base("VTX")
+    {
+        _period = period;
+        _window = new RollingWindow<IBaseDataBar>(period + 1);
+    }
+
+    /// <summary>
+    /// Computes the next value of the Vortex Indicator.
+    /// </summary>
+    /// <param name="input">The input data point.</param>
+    /// <returns>The value of the Vortex Indicator.</returns>
+    protected override decimal ComputeNextValue(IBaseDataBar input)
+    {
+        _window.Add(input);
+
+        if (_window.Count < _period + 1)
+        {
+            return 0m; // or possibly Decimal.MinValue to indicate insufficient data
```

Comment:
As a matter of fact, you should implement `IIndicatorWarmUpPeriodProvider` (since it's period based) but not really `WindowIndicator` but keep two windows for positive and negative vm, along with an ATR indicators as stated in another comment